app/src/main/java/ch/hikemate/app/ui/map/MapScreen.kt

```diff
@@ -363,64 +393,90 @@ fun MapScreen(
     }
   }
 
-  BottomBarNavigation(
-      onTabSelect = { navigationActions.navigateTo(it) },
-      tabList = LIST_TOP_LEVEL_DESTINATIONS,
-      selectedItem = Route.MAP) {
-        Box(modifier = Modifier.fillMaxSize().testTag(Screen.MAP)) {
-          // Jetpack Compose is a relatively recent framework for implementing Android UIs.
-          // OSMDroid
-          // is
-          // an older library that uses Activities, the previous way of doing. The composable
-          // AndroidView
-          // allows us to use OSMDroid's legacy MapView in a Jetpack Compose layout.
-          AndroidView(
-              factory = { mapView },
-              modifier =
-                  Modifier.fillMaxSize()
-                      .testTag(MapScreen.TEST_TAG_MAP)
-                      .padding(bottom = MapScreen.BOTTOM_SHEET_SCAFFOLD_MID_HEIGHT))
-
-          // Button to center the map on the user's location
-          MapMyLocationButton(
-              onClick = {
-                val hasLocationPermission =
-                    LocationUtils.hasLocationPermission(locationPermissionState)
-                // If the user has granted at least one of the two permissions, center the map on
-                // the user's location
-                if (hasLocationPermission) {
-                  MapUtils.centerMapOnUserLocation(context, mapView, userLocationMarker)
-                }
-                // If the user yet needs to grant the permission, show a custom educational alert
-                else {
-                  showLocationPermissionDialog = true
-                }
-              },
-              modifier =
-                  Modifier.align(Alignment.BottomStart)
-                      .padding(bottom = MapScreen.BOTTOM_SHEET_SCAFFOLD_MID_HEIGHT + 8.dp)
-                      .testTag(MapScreen.TEST_TAG_CENTER_MAP_BUTTON))
-          // Search button to request OSM for hikes in the displayed area
-          if (!isSearching.value) {
-            MapSearchButton(
-                onClick = {
-                  MapScreen.launchSearch(isSearching, hikingRoutesViewModel, mapView, context)
-                },
-                modifier =
-                    Modifier.align(Alignment.BottomCenter)
-                        .padding(bottom = MapScreen.BOTTOM_SHEET_SCAFFOLD_MID_HEIGHT + 8.dp))
+  val errorMessageIdState = profileViewModel.errorMessageId.collectAsState()
+  val profileState = profileViewModel.profile.collectAsState()
+
+  when {
```

Comment:
Question: Since you are reusing this pattern twice on the MapScreen aswell as in the HikeDetailsScreen shouldn't this be made into a function? Or is that not possible?

Reply:
I think that is the issue with Sonar.
I will try to refactor it to some kind of function but this might be hard to do.

Reply:
Did you try to do it?
Is it too difficult/lengthy to implement?

Reply:
I just did it!